tests/pyroutils_test.py

```diff
@@ -25,3 +186,206 @@ def prior(num_samples=1):
+        (NLE, ConditionalDensityEstimatorDistribution),
+        (NPE, ConditionalDensityEstimatorDistribution),
+        (NRE, RatioEstimatorDistribution),
+    ],
+)
+def test_estimator_distribution_basic_properties(
+    trainer_cls,
+    distribution_cls,
+    num_dim,
+    num_simulations: int = 100,
+):
+    """Test basic properties of the estimator distribution."""
+    if num_dim == 0:
+        prior = torch.distributions.Normal(0.0, 1.0)
+    else:
+        prior = torch.distributions.MultivariateNormal(
+            loc=torch.zeros(num_dim), covariance_matrix=torch.diag(torch.ones(num_dim))
+        )
+    theta = prior.sample(torch.Size([num_simulations]))
+    x = torch.distributions.Normal(theta, 1.0).sample()
+    trainer = trainer_cls(prior=prior).append_simulations(theta=theta, x=x)
+    density_estimator = trainer.train()
+
+    # Test checking of condition shape
+    with pytest.raises(ValueError):
+        distribution_cls(estimator=density_estimator, condition=theta[:, :1])
+
+    # Test basic properties
+    estimator_dist = distribution_cls(estimator=density_estimator, condition=theta)
+    assert isinstance(estimator_dist, distribution_cls)
+    assert estimator_dist.estimator == density_estimator
+    assert estimator_dist.get_condition_and_event_shapes() == (
+        torch.Size([num_dim]),
+        torch.Size([num_dim]),
+    )
+    assert estimator_dist.batch_shape == torch.Size([num_simulations])
+    assert estimator_dist.event_shape == torch.Size([num_dim])
+    assert estimator_dist.support == torch.distributions.constraints.real
+
+    # Test sample method
+    if trainer_cls is not NRE:
+        assert estimator_dist.sample().shape == torch.Size([num_simulations, num_dim])
+        x_samples = estimator_dist.sample(torch.Size([3]))
+        assert x_samples.shape == torch.Size([3, num_simulations, num_dim])
+    else:
+        with pytest.raises(NotImplementedError):
+            estimator_dist.sample()
+        with pytest.raises(NotImplementedError):
+            estimator_dist.sample(torch.Size([3]))
+        x_samples = torch.distributions.Normal(theta, 1.0).sample(torch.Size([3]))
+
+    # Test log_prob method
+    assert estimator_dist.log_prob(x).shape == torch.Size([num_simulations])
+    assert estimator_dist.log_prob(x_samples).shape == torch.Size([3, num_simulations])
+
+    # Test expand method
+    estimator_dist_expanded = estimator_dist.expand(torch.Size([2, num_simulations]))
+    assert isinstance(estimator_dist_expanded, distribution_cls)
+    assert estimator_dist_expanded.batch_shape == torch.Size([2, num_simulations])
+    assert estimator_dist_expanded.event_shape == estimator_dist.event_shape
+    assert torch.equal(
+        estimator_dist_expanded.condition,
+        theta.expand(torch.Size([2, num_simulations, num_dim])),
+    )
+    assert estimator_dist_expanded.estimator == estimator_dist.estimator
+
+
+@pytest.mark.parametrize("num_subjects", [1])  # unused, required by fixture
```

Comment:
Alternatively unify the two models by renaming `num_dim` -> `num_subjects` or vice versa and this can be removed.

Reply:
I think it's fine to have this param as a placeholder. no need to change it.